File: gpt-4/stage2/parallel_construct_default_clause.cpp
#include <stdlib.h>
#include <openacc.h>
#include "acc_testsuite.h"

#ifndef T1
/*T1:parallel construct default clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);
    const int array_size = 100;
    float *a = (float*) malloc(array_size * sizeof(float));

    #pragma acc data copy(a[0:array_size])
    {
        #pragma acc parallel loop default(present)
        for(int i=0; i<array_size; i++) {
            a[i] = i * 1.0f;
        }
    }

    /* check results */
    for(int i=0; i<array_size; i++) {
        if(a[i] != i*1.0f) {
            err = 1;
            break;
        }
    }

    free(a);
    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/parallel_construct_default_clause.f90
!T1:parallel construct default clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 1000
        REAL(KIND=4), DIMENSION(N) :: a, b, c
        INTEGER :: i

        ! Initialize arrays
        DO i = 1, N
          a(i) = i * 1.0
          b(i) = i * 2.0
        END DO

        !$acc parallel loop copyin(a, b) copyout(c)
        DO i = 1, N
          c(i) = a(i) + b(i)
        END DO
        !$acc end parallel loop

        ! Verify the results
        DO i = 1, N
          IF (c(i) .ne. a(i) + b(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM